src/ln/channelmonitor.rs

```diff
@@ -1670,15 +1814,17 @@ impl ChannelMonitor {
 
 						htlc_timeout_tx.input[0].witness.push(Vec::new()); // First is the multisig dummy
 
-						htlc_timeout_tx.input[0].witness.push(their_sig.serialize_der().to_vec());
+						htlc_timeout_tx.input[0].witness.push(their_sig.clone().serialize_der().to_vec());
 						htlc_timeout_tx.input[0].witness[1].push(SigHashType::All as u8);
-						htlc_timeout_tx.input[0].witness.push(our_sig.serialize_der().to_vec());
+						htlc_timeout_tx.input[0].witness.push(our_sig.clone().serialize_der().to_vec());
 						htlc_timeout_tx.input[0].witness[2].push(SigHashType::All as u8);
 
 						htlc_timeout_tx.input[0].witness.push(Vec::new());
-						htlc_timeout_tx.input[0].witness.push(chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key).into_bytes());
+						let htlc_script = chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &local_tx.a_htlc_key, &local_tx.b_htlc_key, &local_tx.revocation_key);
+						htlc_timeout_tx.input[0].witness.push(htlc_script.clone().into_bytes());
 
 						add_dynamic_output!(htlc_timeout_tx, 0);
+						//TODO: we should delay broadcast of HTLC-timeout tx until locktime - 1 to avoid being bounce off by mempool
```

Comment:
Going through core mempool rules the other day, I read that locktime transactions are going to be rejected from the mempool is locktime isn't equal or more to tip + 1 to avoid DoS vectors which means if we should broadcast timeout txn only at this given height to avoid having to rebuild it few blocks later. And for so track them with an OnchainEvent. It is how the mempool is working ?

Reply:
Ohh, I see your point. Still, broadcast_transaction *should* imply re-announcement (potentially at the appropriate time), if only because mempool transactions may time out. The docs on broadcast_transaction should make sure this is clear, but we shouldn't bother dealing with that.